app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java

```diff
@@ -36,16 +40,48 @@
     private Context mContext;
     private BgGraphBuilder bgGraphBuilder;
     private BgReading mBgReading;
+    private static int lastTransactionId;
+    public PebbleSync(Context context){
+        this.mContext = context;
+        mBgReading = null;
+        init();
+    }
+    private void init() {
+        Log.i(TAG, "Initialising...");
+        Log.i(TAG, "configuring PebbleDataReceiver");
+
+        PebbleKit.registerReceivedDataHandler(mContext, new PebbleKit.PebbleDataReceiver(PEBBLEAPP_UUID) {
+            @Override
+            public void receiveData(final Context context, final int transactionId, final PebbleDictionary data) {
```

Comment:
Pebble watch face triggers this every minute to get the last Bg record/data. Think of it as a keep alive. If it doesn't get a response, it thinks the app is dead. 
I really need to write an xDrip offline pebble app. Neither i, ktind, like the offline watch face. The code sucks and it is doing way too much. 

 
John Stevens
M - +61 418526349

> On 20 May 2015, at 07:52, Stephen Black <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java:
> 
> > @@ -36,16 +40,48 @@
> >      private Context mContext;
> >      private BgGraphBuilder bgGraphBuilder;
> >      private BgReading mBgReading;
> > +    private static int lastTransactionId;
> > +    public PebbleSync(Context context){
> > +        this.mContext = context;
> > +        mBgReading = null;
> > +        init();
> > +    }
> > +    private void init() {
> > +        Log.i(TAG, "Initialising...");
> > +        Log.i(TAG, "configuring PebbleDataReceiver");
> > +
> > +        PebbleKit.registerReceivedDataHandler(mContext, new PebbleKit.PebbleDataReceiver(PEBBLEAPP_UUID) {
> > +            @Override
> > +            public void receiveData(final Context context, final int transactionId, final PebbleDictionary data) {
> so my question is, will this receiveData in between us getting new data on our end? or should this only get triggered once per new incoming BG.
> 
> Also, what exactly triggers this?
> 
> —
> Reply to this email directly or view it on GitHub.
>